src/Core/Domain/Store/ValueObject/StoreId.php

```diff
@@ -0,0 +1,61 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Store\ValueObject;
+
+use PrestaShop\PrestaShop\Core\Domain\Store\Exception\StoreConstraintException;
+
+/**
+ * Contains store ID with it's constraints
+ */
+class StoreId
+{
+	/**
+	 * @var int
+	 */
+	private $storeId;
+
+	/**
+	 * @param int $storeId
+	 */
+	public function __construct(int $storeId)
+	{
+		if (0 >= $storeId) {
+			throw new StoreConstraintException(sprintf('Invalid store id "%d"', $storeId), StoreConstraintException::INVALID_ID);
+		}
```

Comment:
Could you add an unit test for testing the exception ?

Reply:
You will find examples of how to test an exception in existing unit tests, for example here: `tests/Unit/Adapter/Image/ImageValidatorTest.php` line 63

it's pretty straightforward 👍